### PR TITLE
Copter: improve fence pre-arm failure messages

### PR DESCRIPTION
**Pull request by rmackay9**
This PR makes the following changes:

- provides a failure message to the user if the fence pre-arm check fails, "vehicle outside fence" or "fence requires position"
- displays "Fence enabled, need 3D Fix" if the GPS pre-arm check fails and it's required only because the fence is enabled
- shorten "Accelerometers" to "Accels" in pre-arm checks to allow message to fix on the HUD and to be consistent with the "Gyros" failure messages.

### CODE REVIEW

**Comment by rmackay9** [comment]
By the way, I ran this through valgrind as Peter suggested and it didn't seem to complain.

File: ArduCopter/AP_Arming.cpp [hunk 1]
```diff
@@ -159,9 +159,10 @@ bool AP_Arming_Copter::fence_checks(bool display_failure)
 {
     #if AC_FENCE == ENABLED
     // check fence is initialised
-    if (!copter.fence.pre_arm_check()) {
-        if (display_failure) {
-            gcs_send_text(MAV_SEVERITY_CRITICAL,"PreArm: check fence");
+    const char* fail_msg = nullptr;
```
Comment (lucasdemarchi) [inline ArduCopter/AP_Arming.cpp]: Nitpick: should be const char *

Comment (rmackay9) [inline ArduCopter/AP_Arming.cpp]: sure, fixed.  we can squash this commit in when we merge to master.

File: libraries/AC_Fence/AC_Fence.cpp [hunk 2]
```diff
@@ -121,7 +121,7 @@ bool AC_Fence::pre_arm_check() const
     }
 
     // if we have horizontal limits enabled, check inertial nav position is ok
-    if ((_enabled_fences & AC_FENCE_TYPE_CIRCLE)!=0 && !_inav.get_filter_status().flags.horiz_pos_abs && !_inav.get_filter_status().flags.pred_horiz_pos_abs) {
+    if ((_enabled_fences & (AC_FENCE_TYPE_CIRCLE|AC_FENCE_TYPE_POLYGON))!=0 && !_inav.get_filter_status().flags.horiz_pos_abs && !_inav.get_filter_status().flags.pred_horiz_pos_abs) {
```
Comment (khancyr) [inline libraries/AC_Fence/AC_Fence.cpp]: Could you add spaces around | ?

**Comment by OXINARF** [comment]
Love it! We should remove lines 95-97 too since we have a detailed RC being sent already.

**Comment by Pedals2Paddles** [comment]
Awesome. The new message wording is perfect!

**Comment by rmackay9** [comment]
semaphore has failed but it doesn't appear to be a compile error to me at least so merging.

**Pull request merged**